merkledag/ipld.md

```diff
@@ -0,0 +1,464 @@
+    "mode": "0755",
+    "owner": "jbenet"
+  },
+  "cat.jpg": {
+    "mlink": "QmCCC...222"
+    "mode": "0644",
+    "owner": "jbenet"
+  },
+  "doge.jpg": {
+    "mlink": "QmCCC...333",
+    "mode": "0644",
+    "owner": "jbenet"
+  }
+}
+```
+
+or in YML
+
+```yml
+---
+foo:
+  mlink: QmCCC...111
+  mode: 0755
+  owner: jbenet
+cat.jpg:
+  mlink: QmCCC...222
+  mode: 0644
+  owner: jbenet
+doge.jpg:
+  mlink: QmCCC...333
+  mode: 0644
+  owner: jbenet
+```
+
+Though we have new properties in the links that are _specific to this datastructure_, we can still resolve links just fine:
+
+```js
+> ipld cat --json QmCCC...CCC/cat.jpg
+{
+  "data": "\u0008\u0002\u0012��\u0008����\u0000\u0010JFIF\u0000\u0001\u0001\u0001\u0000H\u0000H..."
+}
+
+> ipld cat --json QmCCC...CCC/doge.jpg
+{
+  "subfiles": [
+    {
+      "mlink": "QmPHPs1P3JaWi53q5qqiNauPhiTqa3S1mbszcVPHKGNWRh"
+    },
+    {
+      "mlink": "QmPCuqUTNb21VDqtp5b8VsNzKEMtUsZCCVsEUBrjhERRSR"
+    },
+    {
+      "mlink": "QmS7zrNSHEt5GpcaKrwdbnv1nckBreUxWnLaV4qivjaNr3"
+    }
+  ]
+}
+
+> ipld cat --yml QmCCC...CCC/doge.jpg
+---
+subfiles:
+  - mlink: QmPHPs1P3JaWi53q5qqiNauPhiTqa3S1mbszcVPHKGNWRh
+  - mlink: QmPCuqUTNb21VDqtp5b8VsNzKEMtUsZCCVsEUBrjhERRSR
+  - mlink: QmS7zrNSHEt5GpcaKrwdbnv1nckBreUxWnLaV4qivjaNr3
+
+> ipld cat --json QmCCC...CCC/doge.jpg/subfiles/1
+{
+  "data": "\u0008\u0002\u0012��\u0008����\u0000\u0010JFIF\u0000\u0001\u0001\u0001\u0000H\u0000H..."
+}
+```
+
+But we can't extract the link as nicely as other properties, as links are meant to _resolve through_.
+
+#### Duplicate property keys
+
+Note that having two properties with _the same_ name IS NOT ALLOWED, but actually impossible to prevent (someone will do it and feed it to parsers), so to be safe, we define the value of the path traversal to be _the first_ entry in the serialized representation. For example, suppose we have the object:
+
+```json
+{
+  "name": "J.C.R. Licklider",
+  "name": "Hans Moravec"
+}
+```
+
+Suppose _this_ was the _exact order_ in the _Canonical Format_ (not json, but cbor), and it hashes to `QmDDD...DDD`. We would _ALWAYS_ get:
+
+```sh
+> ipld cat --json QmDDD...DDD
+{
+  "name": "J.C.R. Licklider",
+  "name": "Hans Moravec"
+}
+> ipld cat --json QmDDD...DDD/name
+"J.C.R. Licklider"
+```
+
+
+#### Path Restrictions
+
+There are some important problems that come about with path descriptions in Unix and the web. For a discussion see [this discussion](https://github.com/ipfs/go-ipfs/issues/1710). In order to be compatible with the models and expectations of unix and the web, IPLD explicitly disallows paths with certain path components. **Note that the data itself _may_ still contain these properties (someone will do it, and there are legitimate uses for it). So it is only _Path Resolvers_ that MUST NOT resolve through those paths.** The restrictions are the same as typical unix and UTF-8 path systems:
+
+
+TODO:
+- [ ] list path resolving restrictions
+- [ ] show examples
```

Comment:
@willglynn could you help me fill this out? 

the more concise here the better, but i suspect this section may be a bit large.

see rendered doc here: https://github.com/ipfs/specs/blob/ipld-spec/merkledag/ipld.md